Challenge3/src/agent.ts

```diff
@@ -0,0 +1,55 @@
+import { Finding, HandleBlock, BlockEvent, getEthersProvider, ethers } from "forta-agent";
+import { L1_DAI_TOKEN_ADDRESS, L1_ESCROW_ABI, L1_ARB_ESCROW_ADDRESS, L1_OPT_ESCROW_ADDRESS } from "./constants";
+import { Contract } from "@ethersproject/contracts";
+import { Provider } from "@ethersproject/providers";
+import { l1Finding, getL1DaiLocked, getL2DaiSupply, emitL1Alert, l2Finding } from "./utils";
+
+let chainId: number;
+let l1OptEscrowBalance: ethers.BigNumber = ethers.BigNumber.from(0);
+let l1ArbEscrowBalance: ethers.BigNumber = ethers.BigNumber.from(0);
+let newL1OptEscrowBalance: ethers.BigNumber;
+let newL1ArbEscrowBalance: ethers.BigNumber;
+let l1DaiLocked: ethers.BigNumber;
+let l2DaiSupply: ethers.BigNumber;
+
+export function provideInitialize(provider: ethers.providers.Provider) {
+  return async function initialize() {
+    const network = await provider.getNetwork();
+    chainId = network.chainId;
+  };
+}
+
+export function provideHandleBlock(provider: ethers.providers.Provider): HandleBlock {
+  return async (blEvent: BlockEvent) => {
```

Comment:
```suggestion
  return async (blockEvent: BlockEvent) => {
```
Small suggestion for a bit of a clearer param name. Applicable in other locations as well.